woocommerce/woocommerce
# Avoid requesting product with id -1 while loading

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





We used to request a product with id -1 (`/wp-json/wc/v3/products/-1?context=edit&_locale=user`) while loading the editor.
This PR modifies the `Editor` component to avoid that request.

Closes #45401.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the new product editor at `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
2. Go to Products > Add
3. Open the browser devTools and go to the `Network` tab.
4. Verify the request with product ID `-1` is not being done.

Before

![Screenshot 2024-04-19 at 4 45 37 PM](https://github.com/woocommerce/woocommerce/assets/1314156/a217bca0-932a-4c4a-abd9-a90273bcdc32)



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
